src/js/components/navigation.js

```diff
@@ -129,6 +137,18 @@ navigation = behavior(
         }
       },
     },
+    keydown: {
+      [NAV_PRIMARY]: keymap({ Escape: hideActiveNavDropdown }),
```

Comment:
Switched this from `BODY` to avoid conflicts with other potential functionality.